test/core/gridmapping/test_coords.py

```diff
@@ -121,21 +130,39 @@ def test_1d_xy_coords(self):
         self.assertEqual(('lon', 'lat'), gm.xy_var_names)
         self.assertEqual(('lon', 'lat'), gm.xy_dim_names)
 
+    def test_to_coords(self):
```

Comment:
This is the new test for the parameter reuse_coords=True. Is it sufficient to test for the dtype of the result here because a newly generated set of coordinates would have been float64? If yes, would it have been a solution that the newly created dimensions have dtype float32 as well instead of float64, e.g. by preserving the dtype of the input coordinates? - For some reason the default of reuse_coords is False, so there will be cases where you want to not reuse them.